src/SolverAlsWrmf.cpp

```diff
@@ -51,6 +52,57 @@ arma::Col<T> cg_solver(const arma::Mat<T> &XtX,
   return x;
 }
 
+template <class T>
+arma::Col<T> scd_nnls_solver_explicit(const arma::Mat<T> &X_nnz,
+                                      const arma::Col<T> &confidence,
+                                      T regularization) {
+  const arma::Mat<T> inv = X_nnz * X_nnz.t()
```

Comment:
`inv.diag() += regularization;` should work

Reply:
That doesn't compile unfortunately.

Reply:
That's because of `const`. I've updated your branch